src/transformers/trainer_tf.py

```diff
@@ -650,7 +653,33 @@ def apply_gradients(self, features, labels):
     @tf.function
     def distributed_training_steps(self, batch):
         with self.args.strategy.scope():
-            self.args.strategy.run(self.apply_gradients, batch)
+
+            nb_instances_in_batch = self.compute_nb_instances(batch)
+            inputs = self.get_step_inputs(batch, nb_instances_in_batch)
+
+            self.args.strategy.run(self.apply_gradients, inputs)
+
+    def compute_nb_instance(self, batch):
```

Comment:
I would recommend to rename the functions `_compute_nb_instance` and `_get_step_inputs`